pkg/validation/usage_groups.go

```diff
@@ -169,6 +169,17 @@ type UsageGroupMatchName struct {
 	ResolvedName   string
 }
 
+func (m *UsageGroupMatchName) IsMoreSpecificThan(other *UsageGroupMatchName) bool {
+	if other == nil {
+		return true
+	}
+	return !strings.HasPrefix(m.ConfiguredName, dynamicLabelNamePrefix) && strings.HasPrefix(other.ConfiguredName, dynamicLabelNamePrefix)
```

Comment:
should this be `strings.Contains` instead of `strings.HasPrefix` ? I saw usage group tests has `app/${labels.service_name}` test data.